### PR TITLE
enhancement/15696-indicator-on-indicator

### PR DESCRIPTION
**Pull request by karolkolodziej**
Added support for calculating indicators on another indicator. See #15696.

### CODE REVIEW

**Review by pawelfus — CHANGES_REQUESTED** [review]
Looks good, just one thing: http://jsfiddle.net/BlackLabel/q5hreoLj/1/ - add SMA, then edit SMA -> now you can choose SMA as a base for itself

File: ts/Extensions/Annotations/Popup.ts [hunk 1]
```diff
@@ -1019,25 +1019,22 @@ H.Popup.prototype = {
             selectBox.setAttribute('id', PREFIX + 'select-' + optionName);
 
             // list all series which have id - mandatory for creating indicator
-            chart.series.forEach(function (serie): void {
+            chart.series.forEach(function (series): void {
+                seriesOptions = series.options as any;
```
Comment (raf18seb) [inline ts/Extensions/Annotations/Popup.ts:1023]: You are casting the whole `seriesOptions` variable to `any`, but only `seriesOptions.params` is throwing an error (other `seriesOptions.name` or `seriesOptions.id` are fine.

I would think about how we can get rid of `as any` totally or at least cast to `any` only `options.params` , not the whole variable (like previously, before your changes).

File: ts/Extensions/Annotations/Popup.ts [hunk 2]
```diff
@@ -1019,25 +1019,22 @@ H.Popup.prototype = {
...
 
-                seriesOptions = serie.options;
+                if (seriesOptions.id !== PREFIX + 'navigator-series') {
+                    const seriesName = seriesOptions.params ?
+                        seriesOptions.name || series.name :
+                        seriesOptions.name || seriesOptions.id || '';
```
Comment (raf18seb) [inline ts/Extensions/Annotations/Popup.ts:1028]: Duplicated `seriesOptions.name`.

Isn't it better?:
```js
const seriesName = seriesOptions.name ||
    (seriesOptions as any).params ? series.name : seriesOptions.id || '';
```

**Review by raf18seb — CHANGES_REQUESTED** [review]

**Review by pawellysy — APPROVED** [review]

File: test/cypress/integration/Highcharts/stock-tools.spec.js [hunk 3]
```diff
@@ -34,3 +34,65 @@ describe('Stock Tools', () => {
...
+
+        cy.addIndicator(); // Add SMA indicator with period 20.
+
+        cy.window().then((win) => {
+            const H = win.Highcharts,
+                chart = H.charts[0];
+
+            // Select the first 3m period.
+            chart.xAxis[0].setExtremes(1565098200000, 1565098200000 + 36e5 *24 *90);
+        });
```
Comment (pawelfus) [inline test/cypress/integration/Highcharts/stock-tools.spec.js:73]: Can we use `cy.chart().then(chart => chart.xAxis[0].setExtremes(...)`?

**Review by pawelfus — APPROVED** [review]
Looks good, one question though ;)

**Comment by pawelfus** [comment]
Edit: http://jsfiddle.net/BlackLabel/q5hreoLj/1/ - is is something with my build or is it still broken @karolkolodziej

**Comment by karolkolodziej** [comment]
I guess it might be something on your end...
Add:
Screenshot 2021-08-10 at 11 59 13
Edit: 
Screenshot 2021-08-10 at 11 59 22

**Review by raf18seb — APPROVED** [review]
Thanks!

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
All good, except please resolve the conflicts.

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**